src/renderer/components/wallet/assets/AssetDetails.tsx

```diff
@@ -53,6 +56,22 @@ export const AssetDetails: React.FC<Props> = (props): JSX.Element => {
 
   const [currentPage, setCurrentPage] = useState(1)
 
+  // (Possible) subscription of upgrade tx
+  const [bnbTxSub, setBnbTxSub] = useState<O.Option<Rx.Subscription>>(O.none)
+  // State of upgrade tx
+  const [bnbTxRD, setBnbTxRD] = useState<TxRD>(RD.initial)
+
+  // unsubscribe of possible previous subscription of upgrade tx
+  // It will be called whenever state of `bnbTxSub` changed
+  useEffect(() => {
+    return () => {
+      FP.pipe(
+        bnbTxSub,
+        O.map((sub) => sub.unsubscribe())
+      )
+    }
+  }, [bnbTxSub])
```

Comment:
actually i still can not understand why you have decided to handle subscriptions at the View layer. 
why can not we use a subscriber-function from a service? (you have renamed it to the `subscribeTx`)

Reply:
The state of sending an upgrade tx is needed in `AssetDetails` only (to display it in tx modal). We don't need to store it "globally", because it's not be used anywhere else nor it's updated in a service by any other service (e.g. websocket service etc.) in this case. That was the only reason we have used this state globally before. 

A simple state handling in a component is just we do need here. If you have a case to store it globally, you can still use `subscribeTx`.